src/Plugin/Field/FieldWidget/Select2EntityReference.php

```diff
@@ -53,7 +112,26 @@ public function formElement(FieldItemListInterface $items, $delta, array $elemen
     $element['#target_type'] = $this->getFieldSetting('target_type');
     $element['#selection_handler'] = $this->getFieldSetting('handler');
     $element['#selection_settings'] = $this->getFieldSetting('handler_settings') + ['match_operator' => 'CONTAINS'];
-    $element['#autocreate'] = isset($this->getFieldSetting('handler_settings')['auto_create']) ? $this->getFieldSetting('handler_settings')['auto_create'] : FALSE;
+    if ($this->getFieldSetting('handler_settings')['auto_create']) {
+      $status = TRUE;
+      $definition = $this->entityTypeManager->getDefinition($element['#target_type']);
+      if ($definition->entityClassImplements(EntityPublishedInterface::class)) {
+        $bundle = reset($element['#selection_settings']['target_bundles']);
+        $field_name = $definition->getKey('published');
```

Comment:
Is it possible to use ``::isPublished()`` instead? Since this is provided by EntityPublishedInterface and this has been checked before.

Reply:
Sure, fixed it